### PR TITLE
Rework wording in environment management file

### PR DESCRIPTION
**Pull request by kcpevey**
## Reference Issues or PRs

<s>While I was working on something else, I came across this environment management page and realized that its completely incorrect. This PR removes it. </s>

It wasn't clear which user this page was directed towards. I have reworded it so that it is clear.

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

**Comment by kcpevey** [comment]
I just realized file is actually directed an admin who is about to build the platform - not at end users who are creating an environment while working on the platform. In that case, I will rework the messaging on this page.

File: docs/docs/how-tos/nebari-environment-management.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
 ---
 id: nebari-environment-management
-title: Nebari Environment Management
+title: Setup Environment for Deployment
```
Comment (viniciusdc) [inline docs/docs/how-tos/nebari-environment-management.md:3]: Setup Deployment Environment ?

Comment (viniciusdc) [reply docs/docs/how-tos/nebari-environment-management.md:3]: maybe?

Comment (kcpevey) [reply docs/docs/how-tos/nebari-environment-management.md]: sure thing. Done!

**Review by viniciusdc — APPROVED** [review]

**Pull request merged**